bucket/heroku-cli.json

```diff
@@ -0,0 +1,46 @@
+{
+    "version": "7.26.2",
+    "homepage": "https://devcenter.heroku.com/articles/heroku-cli",
+    "description": "An essential part of using Heroku, which can perform nearly any Heroku-related task right from your terminal.",
+    "license": "ISC",
+    "architecture": {
+        "64bit": {
+            "url": "https://cli-assets.heroku.com/heroku-win32-x64.tar.xz#/dl",
+            "hash": "b552e8c121df572b25a7c96e7272c6eeacdfa49dac0af085ee78b03710b28739"
+        },
+        "32bit": {
+            "url": "https://cli-assets.heroku.com/heroku-win32-x86.tar.xz#/dl",
+            "hash": "b9cab430271936f58f8fc5b4da663307ad6292ab10b6d5a24dfbe642c686518c"
+        }
+    },
+    "depends":"sudo",
+    "bin": "bin/heroku.cmd",
+    "checkver": {
+        "url": "https://cli-assets.heroku.com/version",
+        "jsonpath": "$.version"
+    },
+    "pre_install": [
+        "& (scoop which 7z) x \"$dir\\dl\" -o\"$dir\" | Out-Null",
+        "Invoke-ExternalCommand -FilePath (scoop which 7z) -ArgumentList \"x `\"$dir\\dl~`\" -o`\"$dir\\dlo`\" -bso0 -bd\" -RunAs | Out-Null",
```

Comment:
What does this do? Why this parameters?

Reply:
I try to disable 7z's output. However a window will show though there is no output.

Reply:
Why do you use it this way?
https://github.com/lukesampson/scoop/blob/b9b48182b4d921aa92e9b3e53ab2d2c376d467d7/lib/decompress.ps1#L38
`Expand-7zipArchive \"$dir\\file..\" -Switches = 'all that junk'` if you need sudo do not call scoop directly, but use something like `Get-HelperPath 7zip`

Reply:
Thanks, I'll try it.

Reply:
Do you mean this?

```
"Expand-7zipArchive \"$dir\\dl\" \"$dir\" | Out-Null",
"Invoke-ExternalCommand -FilePath (Get-HelperPath 7zip) -ArgumentList \"x `\"$dir\\dl~`\" -o`\"$dir\\dlo`\" -bso0 -bd\" -RunAs | Out-Null",
```

Reply:
I am still missing why you need to use Expand-7zipArchive on `$fname`? It will be done automatically.

Reply:
If you need privileges this will not work. There is no sudo or anything in first call.

Reply:
Just becaue I need privilege. Is there some way to make scoop decompress download file with privilege?

Reply:
And how does this `"Expand-7zipArchive \"$dir\\dl\" \"$dir\" | Out-Null` help you with extraction with priviledges?

Reply:
It needs to be something like `sudo Expand-7zipArchive \"$dir\\$fname\" | Out-Null`
or again with External command funciton and custom parameters

Reply:
Only need privilege when decompress tar. xz can be decompressed without privilege. Privilege is needed because 7z need to create some syslink.

Reply:
Yes I understand. But if you do not use sudo in line 1 of pre_install it should not work, bucease `Expand-7zipArchive` is not spawning priviledged shell.

Reply:
I don't understand you 😕 but it works.  Expand-7zipArchive doesn't need privilege.

Reply:
If it is working with `Expand-7zipArchive $dir\$fname` in pre_install script it needs to work while normal expanding of renamed file. It is doing complelely same operation. I will look into it as soon as i get home from office.

Reply:
The first line decompress dl(.tar.xz) to dl~(.tar). This step doesn't need privilege. The second line decompress dl~(.tar) which need privilege to create some symbolic link. I rename download file to dl.xz so we don't need the first line now.